lib/command.js

```diff
@@ -225,7 +225,13 @@ module.exports = function command (yargs, usage, validation) {
 
     if (commandHandler.handler && !yargs._hasOutput()) {
       yargs._setHasOutput()
-      commandHandler.handler(innerArgv)
+      const handlerResult = commandHandler.handler(innerArgv)
+      if (handlerResult && typeof handlerResult.then === 'function') {
+        handlerResult.then(
+          null,
+          (error) => yargs.getUsageInstance().fail(null, error)
```

Comment:
I would probably just fail with `err.message` rather than `null`, I think your changes in `usage.js` probably wouldn't be needed then?

Reply:
I'm passing null here and making the changes in `usage.js` so that the error **with stack trace** gets logged to the console, similar to what happens when an error is thrown in a sync command handler.

(On that note, maybe we should align how errors are handled in sync vs async command handlers, and handle sync errors as "failures" too?)

It seems important to have the stack trace logged to the console, and it seems like this is the simplest way to achieve it.

I should probably add a test to make sure this happens. I will add one if you agree that that's what we want to happen.

Reply:
sorry for the slow reply -- this logic seems reasonable to me. Let's add a test for both this and the happy path. To test the happy path, I would just have a command handler that modifies a variable, and assert that it executed as expected:

```js
let asyncHandlerCalled = false
const asyncHandler = new Promise((resolve, reject) => {
  asyncHandlerCalled = true
  return resolve()
})
yargs('foo')
  .command('foo', 'foo command', noop, (yargs) => Promise.reject(error))
```

something like that; you'll need to make sure the test is async, etc.